src/components/ec-table-filter/ec-table-filter.vue

```diff
@@ -0,0 +1,167 @@
+<template>
+  <section
+    class="ec-table-filter"
+    data-test="ec-table-filter"
+  >
+    <component
+      :is="filter.component"
+      v-for="(filter, index) in filters"
+      :key="index"
+      :value="value[filter.name]"
+      v-bind="filter"
+      :label="filter.label"
+      :data-test="`ec-table-filter__filter-item-${index}`"
+      class="ec-table-filter__filter-item"
+      @change="onChange(filter.name, $event)"
+    />
+    <!-- TODO ONL-4893
+    <button
+      type="button"
+      data-test="ec-table-filter__less-filters-button"
+      class="ec-table-filter__less-filters-button"
+      @click="handleLessFilterClick()"
+    >
+      {{ setLessFiltersButtonText }}
+      <ec-icon
+        class="ec-table-filter__less-filters-button--icon"
+        :size="16"
+        :name="setLessFiltersButtonIcon"
+        type="interactive"
+      />
+    </button>
+    -->
+    <button
+      v-if="hasFilters"
+      type="button"
+      data-test="ec-table-filter__clear-filters-button"
+      class="ec-table-filter__clear-filters-button"
+      @click="clearFilters()"
+    >
+      {{ clearFiltersButtonText }}
+    </button>
+  </section>
+</template>
+<script>
+import EcIcon from '../ec-icon';
+
+export default {
+  name: 'EcTableFilter',
+  components: { EcIcon },
+  model: {
+    prop: 'value',
+    event: 'change',
+  },
+  props: {
+    value: {
+      type: Object,
+      default: () => ({}),
+      required: false,
+    },
+    filters: {
+      type: Array,
+      required: true,
+      default: () => ([]),
+    },
+    lessFiltersButtonText: {
+      type: Array,
+      required: false,
+      default: () => (['Less filters', 'More filters']),
+    },
+    clearFiltersButtonText: {
+      type: String,
+      required: false,
+      default: 'Clear filters',
+    },
+    popoverOptions: {
+      type: Object,
+    },
+  },
+  data() {
+    return {
+      isLessFilterActive: true,
+    };
+  },
+  computed: {
+    hasFilters() {
+      return !!Object.keys(this.value).length;
+    },
+    // TODO ONL-4893
+    // setLessFiltersButtonIcon() {
+    //   return this.isLessFilterActive ? 'simple-chevron-down' : 'simple-chevron-up';
+    // },
+    // setLessFiltersButtonText() {
+    //   return this.isLessFilterActive ? this.lessFiltersButtonText[1] : this.lessFiltersButtonText[0];
```

Comment:
Same here, `[1]` and `[0]` are not readable when you compare it to `? this.lessFiltersButtonText : this.moreFiltersButtonText`